gns/train.py

```diff
@@ -165,46 +105,46 @@ def rollout(
     return output_dict, loss
 
 
-def predict(device: str):
+def predict(device: str, cfg: DictConfig):
     """Predict rollouts.
 
     Args:
-      simulator: Trained simulator if not will undergo training.
+      device: 'cpu' or 'cuda'.
+      cfg: configuration dictionary.
 
     """
     # Read metadata
-    metadata = reading_utils.read_metadata(FLAGS.data_path, "rollout")
-    simulator = _get_simulator(metadata, FLAGS.noise_std, FLAGS.noise_std, device)
+    metadata = reading_utils.read_metadata(cfg.data.path, "rollout")
+    simulator = _get_simulator(
+        metadata,
+        cfg.data.num_particle_types,
+        cfg.data.noise_std,
+        cfg.data.noise_std,
+        device,
+    )
 
     # Load simulator
-    if os.path.exists(FLAGS.model_path + FLAGS.model_file):
-        simulator.load(FLAGS.model_path + FLAGS.model_file)
+    if os.path.exists(cfg.model.path + cfg.model.file):
+        simulator.load(cfg.model.path + cfg.model.file)
     else:
-        raise Exception(
-            f"Model does not exist at {FLAGS.model_path + FLAGS.model_file}"
-        )
+        raise Exception(f"Model does not exist at {cfg.model.path + cfg.model.file}")
 
     simulator.to(device)
     simulator.eval()
 
     # Output path
-    if not os.path.exists(FLAGS.output_path):
-        os.makedirs(FLAGS.output_path)
+    if not os.path.exists(cfg.output.path):
+        os.makedirs(cfg.output.path)
 
     # Use `valid`` set for eval mode if not use `test`
     split = (
         "test"
-        if (
-            FLAGS.mode == "rollout"
-            or (not os.path.isfile("{FLAGS.data_path}valid.npz"))
-        )
+        if (cfg.mode == "rollout" or (not os.path.isfile("{cfg.data_path}valid.npz")))
```

Comment:
`"{cfg.data_path}valid.npz"` to `f"{cfg.data_path}valid.npz"`

Reply:
I assume you mean `{cfg.data.path}valid.npz`

Reply:
Shouldn't it be `f"{cfg.data_path}valid.npz"` instead of `"{cfg.data_path}valid.npz"`?